From d734ecafcffcc9384128d19ccce1d49fed028ba4 Mon Sep 17 00:00:00 2001
From: Cavus Mustafa <mustafa.cavus@intel.com>
Date: Tue, 17 Mar 2020 23:48:23 -0700
Subject: [PATCH 1/2] Temporary fix for gcc 9

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e48cee57..d91b06fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,6 +41,8 @@ message (STATUS "CMAKE_BUILD_TYPE ...................... " ${CMAKE_BUILD_TYPE})
 file(REMOVE "${CMAKE_BINARY_DIR}/targets_developer.cmake")
 file(REMOVE "${CMAKE_BINARY_DIR}/targets.cmake")
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=deprecated-declarations")
+
 function(build_ngraph)
     if(NOT ENABLE_NGRAPH)
         return()
-- 
2.17.1


From a6211bfa8f0daed7fb92f8fc8ddee6acec0f4bf7 Mon Sep 17 00:00:00 2001
From: Cavus Mustafa <mustafa.cavus@intel.com>
Date: Wed, 18 Mar 2020 02:00:57 -0700
Subject: [PATCH 2/2] mkldnn and cldnn fix for gcc 9

---
 inference-engine/thirdparty/clDNN/CMakeLists.txt | 3 +++
 inference-engine/thirdparty/mkldnn.cmake         | 3 ++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/inference-engine/thirdparty/clDNN/CMakeLists.txt b/inference-engine/thirdparty/clDNN/CMakeLists.txt
index b08c2744..5f8681d4 100644
--- a/inference-engine/thirdparty/clDNN/CMakeLists.txt
+++ b/inference-engine/thirdparty/clDNN/CMakeLists.txt
@@ -99,6 +99,9 @@ set(CLDNN_BUILD__DEFAULT_OUT_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/build/out")
 # Prefix for all targets in internal pass.
 set(CLDNN_BUILD__PROJ_NAME_PREFIX "")
 
+# Temporary fix for gcc9
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=pessimizing-move -Wno-error=deprecated-copy")
+
 # Single/multi-configuration generator helpers.
 if(CMAKE_CFG_INTDIR STREQUAL ".")
   set(CLDNN__TARGET_CFG_VAR "${CMAKE_BUILD_TYPE}")
diff --git a/inference-engine/thirdparty/mkldnn.cmake b/inference-engine/thirdparty/mkldnn.cmake
index 6d026de5..abaaa1b6 100644
--- a/inference-engine/thirdparty/mkldnn.cmake
+++ b/inference-engine/thirdparty/mkldnn.cmake
@@ -142,4 +142,5 @@ endif()
 endif()
 ## enable jit_gemm from mlk-dnn
 
-target_link_libraries(${TARGET} PRIVATE ${${TARGET}_LINKER_LIBS})
\ No newline at end of file
+target_link_libraries(${TARGET} PRIVATE ${${TARGET}_LINKER_LIBS})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-stringop-overflow")
-- 
2.17.1

